pkg/retriever/coordinators/race_test.go

```diff
@@ -147,11 +147,12 @@ func TestRace(t *testing.T) {
 			childCtx, childCancel := context.WithCancel(ctx)
 			defer childCancel()
 			go func() {
-				retrievalCalls := make([]types.CandidateRetrievalCall, 0, len(testCase.results))
-				for _, result := range testCase.results {
-					retrievalCalls = append(retrievalCalls, types.CandidateRetrievalCall{
-						CandidateRetrieval: &timeoutRetriever{result, childCtx, clock, startChan},
-					})
+				retrievalCalls := func(ctx context.Context, callRetrieval func(types.RetrievalTask)) {
+					for _, result := range testCase.results {
+						callRetrieval(types.AsyncRetrievalTask{
+							AsyncCandidateRetrieval: &timeoutRetriever{result, childCtx, clock, startChan},
```

Comment:
Is this correct to be using `childCtx`? the `ctx` from `Race()` which is used to cancel retrievals when one wins would be the important one to use wouldn't it? I see the same thing in splitretriever.go and will comment there too.

Reply:
yea that's correct. I didn't fix while converting I think.